File: CourceExamples/asynchw/src/main/java/by/it/academy/asynchw/MainActivity.java
package by.it.academy.asynchw;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import java.util.List;
import java.util.concurrent.Executors;

import by.it.academy.asynchw.database.Database;
import by.it.academy.asynchw.repo.DataRepository;
import by.it.academy.asynchw.repo.FutureDataRepository;
import by.it.academy.asynchw.repo.RepositoryListener;
import by.it.academy.asynchw.repo.RxJavaDataRepository;

public class MainActivity extends AppCompatActivity {

    private DataRepository dataRepository;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dataRepository = new FutureDataRepository(new Database(), Executors.newFixedThreadPool(4), getMainExecutor());
        dataRepository = new RxJavaDataRepository(new Database(), Executors.newFixedThreadPool(4), getMainExecutor());

        dataRepository.getAllContacts(new RepositoryListener<List<Contact>>() {
            @Override
            public void onDataReceived(List<Contact> data) {

            }
        });
    }


}

File: CourceExamples/dependencyinjection/src/main/java/by/it/academy/dependencyinjection/MainActivity.kt
package by.it.academy.dependencyinjection

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import javax.inject.Inject
import javax.inject.Named

/**
 * @Inject
 * @Component
 */

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var repo: NewsRepo

    @Inject
    @Named("provideAllNewsRepo1")
    lateinit var allrepo: AllNewsRepoImpl

    @Inject
    lateinit var godRepo: GodRepo

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        (applicationContext as App).appComponent.inject(this)
        repo.getAllNews()
        allrepo.getAllNews()
    }
}
